app/models/twilio_redirection.rb

```diff
@@ -0,0 +1,31 @@
+class TwilioRedirection
+  def self.for(twilio_number)
+    if (location = ::Location.current.find_by(twilio_number: twilio_number))
+      Location.new(location.booking_location || location)
+    elsif (call_centre = ::CallCentre.find_by(twilio_number: twilio_number))
+      CallCentre.new(call_centre)
+    end
+  end
+
+  class Location
+    def initialize(location)
+      @location = location
+    end
+
+    def phone
+      return ::Location::TP_CALL_CENTRE_NUMBER if @location.hidden?
```

Comment:
How do you feel about giving the guard conditions and following statements a line of whitespace for breathing room?